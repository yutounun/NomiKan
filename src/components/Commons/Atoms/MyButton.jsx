import PropTypes from 'prop-types'
import React from 'react'
import Button from '@mui/material/Button'

const MyButton = (props) => {
  return (
    <Button
      variant={props.variant}
      color={props.color}
      size={props.size}
      startIcon={props.startIcon}
    >
      {props.value}
    </Button>
  )
}

MyButton.propTypes = {
  variant: PropTypes.string,
  value: PropTypes.string,
  color: PropTypes.string,
  size: PropTypes.string,
  startIcon: PropTypes.node,
}
export default MyButton
