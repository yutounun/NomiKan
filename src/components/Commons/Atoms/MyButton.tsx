import Button from '@mui/material/Button'

interface Props {
  variant?: 'contained' | 'outlined'
  value?: string
  color?:
    | 'inherit'
    | 'primary'
    | 'secondary'
    | 'success'
    | 'error'
    | 'info'
    | 'warning'
  size?: 'small' | 'medium' | 'large'
  disabled?: boolean
  onClick?: () => void
}
const MyButton: React.FC<Props> = (props) => {
  return (
    <Button
      onClick={props.onClick}
      data-testid='button'
      variant={props.variant}
      color={props.color}
      size={props.size}
      disabled={props.disabled}
    >
      {props.value}
    </Button>
  )
}
export default MyButton
