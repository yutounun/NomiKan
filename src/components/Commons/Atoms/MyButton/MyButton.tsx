import React from 'react'
import Button from '@mui/material/Button'

type Props = {
  variant: 'text' | 'contained' | 'outlined' | undefined
  value: string
  color:
    | 'inherit'
    | 'primary'
    | 'secondary'
    | 'success'
    | 'error'
    | 'info'
    | 'warning'
    | undefined
  size: 'small' | 'medium' | 'large' | undefined
  startIcon: React.ReactNode
}

const MyButton: React.FC<Props> = (props) => {
  return (
    <Button
      variant={props.variant}
      color={props.color}
      size={props.size}
      startIcon={props.startIcon}
    >
      {props.value}
    </Button>
  )
}
export default MyButton
