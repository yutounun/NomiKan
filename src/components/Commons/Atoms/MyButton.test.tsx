import { render, screen } from "@testing-library/react";
import React from "react";
import MyButton from "./MyButton";

describe("MyButton", () => {
  it("renders correctly", () => {
    render(<MyButton dataTestid="button" />);
    expect(screen.getByTestId("button")).toBeInTheDocument();
  });
  it("takes a snapshot", () => {
    const { asFragment } = render(<MyButton />);
    expect(asFragment()).toMatchSnapshot();
  });
});
