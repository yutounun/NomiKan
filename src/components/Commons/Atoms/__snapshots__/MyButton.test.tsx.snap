// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyButton takes a snapshot 1`] = `
<DocumentFragment>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1e6y48t-MuiButtonBase-root-MuiButton-root"
    data-testid="button"
    style="width: 200px;"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
</DocumentFragment>
`;
